src/main/java/hudson/plugins/sshslaves/verifiers/ManuallyProvidedKeyVerificationStrategy.java

```diff
@@ -92,18 +93,36 @@ private static HostKey parseKey(String key) {
                 RSASHA1Verify.decodeSSHRSAPublicKey(keyValue);
             } else if ("ssh-dss".equals(algorithm)) {
                 DSASHA1Verify.decodeSSHDSAPublicKey(keyValue);
+            } else if (isClass("com.trilead.ssh2.signature.ED25519KeyAlgorithm") && "ssh-ed25519".equals(algorithm)) {
```

Comment:
I would reorder operations. `isClass()` is much more heavy than String comparison